### PR TITLE
fix(core): toolbar bar horizon updates

### PR DESCRIPTION
**Pull request by mikerodonnell89**
part of #7931 

Adds the option to simply pass a title via string input, or the developer can use the existing method of passing via ng-content

### CODE REVIEW

**Comment by g-cheishvili** [comment]
So, `When the text inside a Toolbar is set to h4, the [hasTitle]="true" must be used as seen in the exmaple below.`, meaning hasTitle should always be applied or only in case of h4? Can't `ToolbarComponent` check it by itself if it contains fd-title?

**Review by N1XUS — APPROVED** [review]

**Comment by mikerodonnell89** [comment]
> So, `When the text inside a Toolbar is set to h4, the [hasTitle]="true" must be used as seen in the exmaple below.`, meaning hasTitle should always be applied or only in case of h4? Can't `ToolbarComponent` check it by itself if it contains fd-title?

@g-cheishvili thanks for pointing this out, I didn't look through the docs

**Comment by mikerodonnell89** [comment]
> So, `When the text inside a Toolbar is set to h4, the [hasTitle]="true" must be used as seen in the exmaple below.`, meaning hasTitle should always be applied or only in case of h4? Can't `ToolbarComponent` check it by itself if it contains fd-title?

I think the docs are implying that the dev should use h4 in the event that they'd like to use a title. In my opinion this could simply be a string input - I have committed a change for this, wondering if anyone has thoughts @InnaAtanasova @g-cheishvili @N1XUS

**Comment by g-cheishvili** [comment]
@mikerodonnell89 that will limit capabilities. what if user wants to have icon in header or have inline help, or anythin. if its input funtionality is limited

**Comment by mikerodonnell89** [comment]
> @mikerodonnell89 that will limit capabilities. what if user wants to have icon in header or have inline help, or anythin. if its input, funtionality is limited

@g-cheishvili The previous docs and design specs for both horizon and fiori 3 imply to me that the toolbar is to have a text-only title. But I suppose going this route might be too opinionated, which goes against the idea of fundamental-ngx/core. Thoughts @InnaAtanasova ?

**Comment by mikerodonnell89** [comment]
@InnaAtanasova has suggested allowing both a simple input or allowing more complex templates via ng-content so I will do that and document both

File: libs/core/src/lib/toolbar/toolbar.component.ts [hunk 1]
```diff
@@ -118,6 +125,10 @@ export class ToolbarComponent implements OnInit, AfterViewInit, OnDestroy, After
     @ContentChildren(forwardRef(() => ToolbarItemDirective))
     toolbarItems: QueryList<ToolbarItemDirective>;
 
+    /** @hidden */
+    @ContentChild(TitleComponent)
+    titleComponent: TitleComponent;
```
Comment (platon-rov) [inline libs/core/src/lib/toolbar/toolbar.component.ts:130]: Querying like this prevent from tree-shaking `TitleComponent` when compiling this component.

Here some more about it https://angular.io/guide/lightweight-injection-tokens

File: libs/core/src/lib/toolbar/toolbar.component.ts [hunk 2]
```diff
@@ -206,6 +217,14 @@ export class ToolbarComponent implements OnInit, AfterViewInit, OnDestroy, After
         this.buildComponentCssClass();
     }
 
+    /** @hidden */
+    ngAfterContentInit(): void {
+        if (this.titleComponent) {
+            this.titleComponent.elementRef?.nativeElement.classList.add('fd-toolbar__title');
```
Comment (platon-rov) [inline libs/core/src/lib/toolbar/toolbar.component.ts:223]: I think it should be done in `TitleComponent` itself. 

There we may get toolbar as an optional dependency and check if it's present - apply class. Also, this dependency should rely not on the implementation but on the injection token, see my comment above.

Comment (mikerodonnell89) [reply libs/core/src/lib/toolbar/toolbar.component.ts:223]: It's a toolbar class so in my opinion makes more sense to happen here

File: libs/core/src/lib/toolbar/toolbar.component.ts (hunk 1 above)
Comment (mikerodonnell89) [reply libs/core/src/lib/toolbar/toolbar.component.ts]: done

**Review by platon-rov — APPROVED** [review]

**Review by InnaAtanasova — APPROVED** [review]

**Comment by InnaAtanasova** [comment]
I noticed the new components that go in the overflow menu don't have a margin between them:
Screen Shot 2022-05-20 at 2 00 17 PM

But I guess it's out of the scope for this PR and will go to the defect hunting issue (or separately).

**Comment by InnaAtanasova** [comment]
> I noticed the new components that go in the overflow menu don't have a margin between them: Screen Shot 2022-05-20 at 2 00 17 PM
> 
> But I guess it's out of the scope for this PR and will go to the defect hunting issue (or separately).

Linked the issue here: https://github.com/SAP/fundamental-ngx/issues/8119

**Pull request merged**